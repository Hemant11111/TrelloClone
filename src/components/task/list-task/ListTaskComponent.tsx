import React from "react";
import { useSelector } from "react-redux";

import "./ListTask.scss";
import TaskModel from "../../../model/Task";
import TaskComponent from "../task-details/TaskComponent";
import AddTaskComponent from "../add-task/AddTaskComponent";
import { getTaskListByCardId } from "../../../redux/selectors/task.selector";

interface ListCardComponentProps {
    cardId: string
}

function ListTaskComponent(props: ListCardComponentProps) {

    const tasks = useSelector(getTaskListByCardId(props.cardId));

    return (
        <div className="task-list-container">
            {
                tasks.map((task: TaskModel) => {
                    return (
                        <TaskComponent
                            key={`task-${task.id}`}
                            task={task}
                        />
                    )
                })
            }

            <AddTaskComponent cardId={props.cardId}/>
        </div>
    )
}

export default React.memo(ListTaskComponent);