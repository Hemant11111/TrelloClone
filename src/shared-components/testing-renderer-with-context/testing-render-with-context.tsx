import { ReactElement } from "react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";

import store from "../../redux/store/store";

export function testingRenderWithContext(element: ReactElement) {
    return render(
        <Provider store={store}>
            {element}
        </Provider>
    )
}