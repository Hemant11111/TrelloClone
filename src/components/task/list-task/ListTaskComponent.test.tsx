import React from 'react';

import ListTaskComponent from "./ListTaskComponent";
import { testingRenderWithContext } from "../../../shared-components/testing-renderer-with-context/testing-render-with-context";


describe('List task Component', () => {

    it('should match snapshot', () => {
        const {asFragment} = testingRenderWithContext(<ListTaskComponent cardId={"1"}/>);
        expect(asFragment()).toMatchSnapshot();
    })

})