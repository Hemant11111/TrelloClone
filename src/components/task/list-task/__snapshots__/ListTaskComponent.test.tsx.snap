// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List task Component should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="task-list-container"
  >
    <div
      class="add-task-container"
    >
      <form
        class="custom-form"
      >
        <input
          class=" input-form"
          placeholder="Add Task..."
          value=""
        />
      </form>
      <button
        class="add-task-button trello-icon-button"
        data-testid="add-task-button"
        disabled=""
      >
        <i
          class="fa fa-plus"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;
